Network/HTTP/Auth.hs

```diff
@@ -36,6 +36,11 @@ import Data.Char
 import Data.Maybe
 import Data.Word ( Word8 )
 
+#include <ghcplatform.h>
+#ifdef HaLVM_TARGET_OS
+#define MIN_VERSION_network(a,b,c) 1
```

Comment:
Option 2 sounds good to me.

I'm open to 3 in principle, but I think it would mean dropping GHC 7.0 support - the last time I asked about that two years ago there were some statistics showing it was still in significant use. It's probably getting time to ask again.